pkg/asset/installconfig/aws/metadata.go

```diff
@@ -250,7 +249,27 @@ func (m *Metadata) PublicSubnets(ctx context.Context) (Subnets, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving Public Subnets: %w", err)
 	}
-	return m.publicSubnets, nil
+	return m.subnets.Public, nil
+}
+
+// Subnets retrieves a group of subnet metadata that is indexed by subnet ID for all provided subnets.
+// This includes private, public and edge subnets.
+func (m *Metadata) Subnets(ctx context.Context) (SubnetGroups, error) {
+	err := m.populateSubnets(ctx)
+	if err != nil {
+		return m.subnets, fmt.Errorf("error retrieving all Subnets: %w", err)
+	}
+	return m.subnets, nil
+}
+
+// VPCSubnets retrieves a group of all subnet metadata that is indexed by subnet ID
+// in the VPC of the provided subnets.
```

Comment:
Having VPCSubnets in the metadata concerns me that it could be used accidentally when intending to use Subnets. The only purpose is for the validation right? Can you add a comment indicating that this may potentially include non-cluster subnets, and that Subnets should be used to select cluster subnets.

Reply:
Sure, good idea! This field is indeed only for validations. Updated, thanks!

Reply:
What did you decide as the resolution? The update is not obvious to me.

Reply:
I added the following comment:

```
// vpcSubnets includes all subnets in the VPC.
// That is cluster subnets, which are provided in the install config and potentially other non-cluster subnets.
// Field subnets should be used to select only cluster subnets.
vpcSubnets    SubnetGroups
```

Maybe, I should also mention that it is only used for validation.

Reply:
How does this https://github.com/openshift/installer/pull/9599/commits/50bbb98197edb5d58750c474047b2a4080bcefcf sound?

Reply:
It sounds good, but I was thinking the comment would be on the func VPCSubnets as that is expected way for devs to interact with the metadata.

Reply:
Lol now it makes sense why it was not obvious to you. My brain was not braining. Let me update it :sweat_smile:

Reply:
Updated in https://github.com/openshift/installer/pull/9599/commits/3da57f1066b10d3bacd7cd674e03a532ec8033cf along with other comments :D